src/scripts/extensions/authenticationHelper.ts

```diff
@@ -71,6 +72,11 @@ export class AuthenticationHelper {
 				if (isValidUser) {
 					const dataBoundaryHelper = new UserDataBoundaryHelper();
 					let userDataBoundary: string = await dataBoundaryHelper.getUserDataBoundary(response.data);
+					// The default logging has been configured to EU Pipeline. Once we find the
+					// userdataboundary and if it is different from EUDB , reinit the logger with WW Pipeline
+					if (userDataBoundary === DataBoundary[DataBoundary.GLOBAL] || userDataBoundary === DataBoundary[DataBoundary.PUBLIC]) {
+						LogManager.reInitLoggerForDataBoundaryChange();
```

Comment:
Does reinitializing mean we will log all the logs in buffer to the new region or partly to EU and partly to WW if data boundary in non-EU?

Reply:
The logs that are captured till this point are already logged to EU region. Once we determine the region, if it is Non EU, then we log to WW else it will be logged to EU.
This way we also try to solve for the case where in before sign in happens , the default logging is going to happen in EU only

Reply:
So with your first statement if the user belong to WW, then the logs before signin goes to EU region and after sigin they go to WW region? Not sure how we will correlate and track the funnel in this case

Reply:
Yes, the logs go to EU region before signing in and after signing in, depending upon boundary they go to respective region. To correlate and track the funnel , i used the DeviceInfo_Id which is common for the session